### PR TITLE
Adding setup configurations 

### LINKED ISSUE #120: Add proper setup configuration
To add `pyproject.toml` and `setup.cfg`.

### PR DESCRIPTION
**Pull request by alexnguyenn**
Closes: #120

Added `pyproject.toml, setup.cfg, setup.py, flake8.ini` to the directory.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+profile = "black"
+multi_line_output = 3
+src_paths = ["lsqecc", "test"]
+
+[tool.black]
+line-length = 100
+
+[tool.mypy]
+ignore_missing_imports = True
+exclude = debug/
```
Comment (gwwatkin) [inline pyproject.toml]: Wonder why this looks weird. Also I do hope to see `debug/` gone, but I don't think we are there yet

File: setup.cfg [hunk 2]
```diff
@@ -0,0 +1,37 @@
+[metadata]
+name = lsqecc
+version = 0.1.0
+description = a Python lattice-surgery based quantum error correction compiler
+long_description = file: README.md
+long_description_content_type = text/markdown; charset=UTF-8
+url = https://latticesurgery.com/
+author = George Watkins, Alex Nguyen, Varun Seshardi, Keeland Watkins
```
Comment (gwwatkin) [inline setup.cfg]:
```suggestion
author = George Watkins, Alex Nguyen, Varun Seshadri, Keelan Watkins
```

**Review by gwwatkin — APPROVED** [review]
Looks good. Minor fixes to the names. 

Still have to familiarize with the new build tools though.

File: pyproject.toml (hunk 1 above)
Comment (alexnguyenn) [inline pyproject.toml]:
```suggestion
ignore_missing_imports = true
```

Comment (alexnguyenn) [reply pyproject.toml]: Just fixed it. I think testing is implemented we will look to remove `debug/` (maybe with some jupyter notebooks to demonstrate some example use cases of the tool).

Comment (gwwatkin) [reply pyproject.toml]: Yeah that's a great idea ... have some notebooks like they do in qiskit

File: pyproject.toml [hunk 3]
```diff
@@ -11,5 +11,4 @@ src_paths = ["lsqecc", "test"]
 line-length = 100
 
 [tool.mypy]
-ignore_missing_imports = True
-exclude = debug/
```
Comment (gwwatkin) [inline pyproject.toml]: Did mypy run fine in the debug folder?

**Pull request merged**